pkg/admission/server.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Serve the http request
+func Serve(w http.ResponseWriter, r *http.Request, admit AdmitFunc) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	// verify the content type is accurate
+	contentType := r.Header.Get(CONTENTTYPE)
+	if contentType != APPLICATIONJSON {
+		glog.Errorf("contentType=%s, expect application/json", contentType)
+		return
+	}
+
+	var reviewResponse *v1beta1.AdmissionResponse
+	ar := v1beta1.AdmissionReview{}
+	deserializer := Codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(body, nil, &ar); err != nil {
+		reviewResponse = ToAdmissionResponse(err)
+	} else {
+		reviewResponse = admit(ar)
+	}
+	glog.V(3).Infof("sending response: %v", reviewResponse)
+
+	response := createResponse(reviewResponse, &ar)
+	resp, err := json.Marshal(response)
+	if err != nil {
+		glog.Error(err)
```

Comment:
when error should directly return something like 
		http.Error(w, fmt.Sprintf("could encode response: %v", err), http.StatusInternalServerError)
                return

Reply:
oh, the part just copy volcano.sh\volcano\cmd\admission\app\server.go, 

so if need fix, new pr do it.